sdk/identity/identity/src/credentials/authFileCredential.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { AccessToken, TokenCredential, GetTokenOptions } from "@azure/core-http";
+import { createSpan } from "../util/tracing";
+import { AuthenticationErrorName } from "../client/errors";
+import { CanonicalCode } from "@opentelemetry/types";
+import { ClientCertificateCredential } from "./clientCertificateCredential";
+import * as fs from "fs";
+
+/**
+ * Enables authentication to Azure Active Directory using client secret
+ * details configured in the following environment variables:
+ *
+ * - AZURE_TENANT_ID: The Azure Active Directory tenant (directory) ID.
+ * - AZURE_CLIENT_ID: The client (application) ID of an App Registration in the tenant.
+ * - AZURE_CLIENT_SECRET: A client secret that was generated for the App Registration.
+ *
+ * This credential ultimately uses a {@link ClientSecretCredential} to
+ * perform the authentication using these details.  Please consult the
+ * documentation of that class for more details.
+ */
+export class AuthFileCredential implements TokenCredential {
+  private credential?: TokenCredential = undefined;
+  /**
+   * Creates an instance of the authFileCredential class.
+   *
+   * @param filePath The path to the SDK Auth file.
+   * @param options Options for configuring the client which makes the authentication request.
+   */
+  constructor(filePath: string) {
+    let authData = JSON.parse(fs.readFileSync(filePath).toString());
```

Comment:
I think this I/O should be moved to `GetToken` where it can be done asynchronously rather than blocking in the constructor.